alacritty/src/display.rs

```diff
@@ -171,16 +172,16 @@ impl Display {
         let metrics = GlyphCache::static_metrics(config.ui_config.font.clone(), estimated_dpr)?;
         let (cell_width, cell_height) = compute_cell_size(config, &metrics);
 
-        let dimensions = GlyphCache::calculate_dimensions(
-            &config.ui_config.window,
-            estimated_dpr,
-            cell_width,
-            cell_height,
-        );
+        // Guess the target window size if the user has specified the number of lines/columns.
+        let dimensions = config.ui_config.window.dimensions();
+        let estimated_size = dimensions.map(|dimensions| {
+            let (padding_x, padding_y) = scale_padding(config, estimated_dpr);
+            window_size(dimensions, padding_x, padding_y, cell_width, cell_height)
+        });
```

Comment:
I wonder why we can't compute window size and set its dimensions in window.rs? We're getting a size letter from calling `inner_size` anyway.

Reply:
I'm not sure what benefit you hope to get? We need to compute window_size in display for several purposes anyways, why do you want to move it?

What do you mean by getting "a size letter" from calling `inner_size`? The size we get from winit is the real size, that's not the goal here. The goal is to get the *target* size.

Reply:
I mean that the only thing you do here is passing it into Window::new, where I suggest to compute it in Window::new passing less arguments into it. You can use window_size there, it's not really a problem, since it's a free function.

So I suggest to remove `estimated_size` from here, remove it from `Window::new` and compute it there.

Reply:
> I mean that the only thing you do here is passing it into Window::new

That is incorrect? We use it to update the window size if the DPR isn't estimated correctly, for which we need the `QuadRenderer`. Are you suggesting we move the `QuadRenderer` to window too? How are we going to know the cell size then to get our SizeInfo?

Reply:
>That is incorrect? We use it to update the window size if the DPR isn't estimated correctly, for which we need the QuadRenderer. Are you suggesting we move the QuadRenderer to window too? How are we going to know the cell size then to get our SizeInfo?

We use `estimated_size` size just to pass it to window, that's it, you can clearly see at the code that you just create it, log it, and move to window::new, so I suggest computing that size inside of window::new, does it make sense?

Reply:
No, that doesn't address the `QuadRenderer` problem?

Reply:
I'll post a patch.